invoices/update.go

```diff
@@ -182,12 +182,14 @@ func updateMpp(ctx *invoiceUpdateCtx,
 	if ctx.mpp.TotalMsat() < inv.Terms.Value {
 		return nil, ctx.failRes(ResultHtlcSetTotalTooLow), nil
 	}
-
 	htlcSet := inv.HTLCSet(setID, channeldb.HtlcStateAccepted)
 
+	// Check whether this invoice is hold invoice or the channel is fake.
+	hodlInvoice := inv.HodlInvoice || ctx.circuitKey.ChanID.IsFake()
```

Comment:
Similar comment here re leaking in details where it may not be necessary. Rather than force the invoice registry to understand zero conf channels, we can instead have the caller (the link in this case) pass along that information by just setting `HoldInvoice` to true.

Reply:
The link doesn't pass the invoice, just the hash. Maybe just add `Hold` field to this function called by the link?
https://github.com/lightningnetwork/lnd/blob/master/invoices/invoiceregistry.go#L872